### ISSUE TITLE
support optional file type in the JSON key

### ISSUE THREAD

**Issue by yairEO** [briangonzalez/gulp-file-contents-to-json#3]
I don't like it that the JSON keys all have the file name  + its extension (type).

I hacked it in my copy of your code, but it's very bad changing the source code, because now we're not synced.

```
// .. index.js .. line 43
var id = file.path.replace(file.base, '').split('/').join(':');   // 'foo/bar/bax.txt' => 'foo:bar:baz.txt'
// now I added the below line which removed the DOT extension
id = id.split('.').slice(0, -1).join('.'); 
```

It would be nice to have this configurable within the gulp setup.

Thanks!

**Comment by briangonzalez** [comment]
So you'd prefer the extensions to be omitted -- how do you resolve clashes?

```
/foo/bar/baz.txt
/foo/bar/baz.md
```

**Comment by yairEO** [comment]
I use this plugin to convert templates files (`.html`) to an Object, and then my app architecture does more with that JSON output, but for me, all files are of type `.html` and I have to create a wrapper function which adds `.html` to every call to the JSON output of the plugin. Would be helpful (cleaner) if there was an optional setting to pass through the GULP plugin.

Of course when doing things like the example you gave, you wouldn't toggle the setting off.

---

I will post a sample To-Do app today or tomorrow so you could see how coll everything is set up and how helpful your plugin is.

**Comment by briangonzalez** [comment]
Want to take a stab and submit a PR? I can help.

**Comment by yairEO** [comment]
I don't know how GULP plugins are authored..
never made any or read. I just hacked the lines I needed,
Is it a lot of work to have an optional argument?

**Comment by briangonzalez** [comment]
No, but it's an best practice for consumers of a library to give back :) 

I'd rather teach you then do it myself.
